dandiapi/api/user_migration.py

```diff
@@ -0,0 +1,39 @@
+from allauth.account.signals import user_logged_in
+from django.dispatch import receiver
+
+
+def copy_ownership(placeholder_user, user):
+    """Copy dandiset ownership from a placeholder user to the real user."""
+    from dandiapi.api.models import Dandiset
+    from guardian.shortcuts import get_objects_for_user, assign_perm, remove_perm
+
+    owned_dandisets = get_objects_for_user(placeholder_user, 'owner', Dandiset)
+    print(f'{placeholder_user} owns {owned_dandisets}')
+    for dandiset in owned_dandisets:
+        print(f'Moving ownership on {dandiset.identifier}')
+        assign_perm('owner', user, dandiset)
+        remove_perm('owner', placeholder_user, dandiset)
+
+
+def depose_placeholder(user):
+    """Replace a placeholder user with a real user, if a placeholder exists."""
+    from django.contrib.auth.models import User
+
+    placeholder_email = 'placeholder_' + user.email
+    try:
+        placeholder_user = User.objects.get(email=placeholder_email)
+    except User.DoesNotExist:
+        # No placeholder user, nothing to do
+        return
+
+    print(f'Replacing {placeholder_user} with {user}')
+    copy_ownership(placeholder_user, user)
```

Comment:
If this step fails partway through for some reason (like, network failure, etc.), will that cause this script to properly abort so we don't end up deleting the placeholder user prematurely?

Reply:
Any python exception should interrupt execution, yes. I'm reasonably confident that all operations are idempotent, but I don't want to spend too much time proving it for these functions that will hopefully only be run once.